pycbc/events/significance.py

```diff
@@ -0,0 +1,253 @@
+# Copyright (C) 2022 Gareth Cabourn Davies
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the
+# Free Software Foundation; either version 3 of the License, or (at your
+# option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+#
+# =============================================================================
+#
+#                                   Preamble
+#
+# =============================================================================
+#
+
+"""
+This module contains functions to calculate the significance
+through different estimation methods of the background, and functions that
+read in the associated options to do so.
+"""
+import logging
+import copy
+import numpy as np
+from pycbc.events import coinc
+from pycbc.events import trigger_fits as trstats
+
+
+def n_louder(bstat, fstat, dec, **kwargs):  # pylint:disable=unused-argument
+    """
+    Count the number of louder background events, taking decimation
+    into account
+    """
+    return coinc.calculate_n_louder(bstat, fstat, dec)
+
+
+def trig_fit(back_stat, fore_stat, dec_facs, fit_func=None,
+             fit_thresh=None):
+    """
+    Use a fit to events in back_stat in order to estimate the
+    distribution for use in recovering the ifars. Below the
+    fit threshold, use the n_louder method for these triggers
+    """
+    # Manually set some defaults:
+    fit_func = fit_func if fit_func else 'exponential'
```

Comment:
is this preferable to doing `fit_func='exponential'` and `fit_thresh=0` in the function definition - that is the 'obvious' way of having defaults, does it not work? 
Also one should be very careful about what values of fit_thresh evaluate to True or False .. ie if the 'if' statement is kept it should be something like 'if fit_thresh != None'